src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -473,7 +482,9 @@ def domain_union(domains: list[SymbolicDomain]) -> SymbolicDomain:
             lambda current_expr, el_expr: im.call("maximum")(current_expr, el_expr),
             [domain.ranges[dim].stop for domain in domains],
         )
-        new_domain_ranges[dim] = SymbolicRange(start, stop)
+        new_domain_ranges[dim] = SymbolicRange(
```

Comment:
Is this change required? I hope not then we shouldn't do it here to avoid creating an unnecessary dependency between the passes. If a cleaner IR is required for tests / debugging this should be done outside of the pass.

Reply:
Yes, it is required to get rid of minimum/maximum in the SymbolicDomains returned by infer_as_fieldop. Should the ConstantFolding be done directly in the tests in the end or somewhere else?

Reply:
As discussed, I removed the ConstantFolding from domain_union and do it now directly in the tests.